libslide/src/scanner.rs

```diff
@@ -0,0 +1,233 @@
+mod types;
+use types::TokenType;
+use types::Token;
+
+pub struct Scanner {
+    input: String, 
+    pub output: Vec<Token>
+}
+
+impl Scanner {
+    // instantiate a new scanner
+    pub fn new(input: &str) -> Scanner {
+        Scanner{
+            input: input.to_owned(),
+            output: Vec::new()
+        }
+    }
```

Comment:
```suggestion
    pub fn new<T: Into<String>>(input: T) -> Scanner {
        Scanner{
            input: input.into(),
            output: Vec::new()
        }
    }
```

This will allow passing a `&str`, `String`, or anything else